src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/CastParser.java

```diff
@@ -48,7 +49,7 @@ public String parse(boolean isReturn) throws SqlParseException {
 
         String dataType = castExpr.getDataType().getName().toUpperCase();
         String fileName = String.format("doc['%s'].value", Util.expr2Object(castExpr.getExpr()));
-        String name = SQLFunctions.randomize("field");
+        String name = "field" + (++fieldId);
```

Comment:
Is it possible this name conflict with what generated in SQLFunctions? Because I notice we do have a method called field.

Reply:
I think we need to make sure either they have different namespace (prefix) or ideally maybe a global incremental counter per request.

Reply:
I wasn't able to come up with the example that could break it, and, in reality, the risk was always there, because we do not track generated identifiers.

Reply:
Technically it's already close to it, because we have one SqlParser per request (see my answer to @penghuo). I like the suggestion of different prefix, implemented

Reply:
I think my question is a little different. I can understand we create new instance per request. But essentially you're using 2 different counters (both starts from 0) with potential conflicting prefixes ("field" and method name which could be "field" too).

Reply:
>  potential conflicting prefixes ("field" and method name which could be "field" too).

That is fixed in last change